src/hotspot/share/cds/filemap.cpp

```diff
@@ -2480,28 +2480,28 @@ void FileMapInfo::stop_sharing_and_unmap(const char* msg) {
 ClassPathEntry** FileMapInfo::_classpath_entries_for_jvmti = NULL;
 
 ClassPathEntry* FileMapInfo::get_classpath_entry_for_jvmti(int i, TRAPS) {
+  if (i == 0) {
+    // index 0 corresponds to the ClassPathImageEntry which is a globally shared object
+    // and should never be deleted.
+    return ClassLoader::get_jrt_entry();
+  }
   ClassPathEntry* ent = _classpath_entries_for_jvmti[i];
   if (ent == NULL) {
-    if (i == 0) {
-      ent = ClassLoader::get_jrt_entry();
-      assert(ent != NULL, "must be");
+    SharedClassPathEntry* scpe = shared_path(i);
+    assert(scpe->is_jar(), "must be"); // other types of scpe will not produce archived classes
+
+    const char* path = scpe->name();
+    struct stat st;
+    if (os::stat(path, &st) != 0) {
```

Comment:
Do you check if the file exists? If so please use os::file_exists which is better.

Reply:
Using `os::file_exists` won't work because few lines below that I need to use the `st` returned from `os::stat`.

`ent = ClassLoader::create_class_path_entry(THREAD, path, &st, false, false);`